test/io/formatters/Json.chpl

```diff
@@ -0,0 +1,231 @@
+
+module Json {
+  private use IO;
+  private use CTypes;
+  private use Map;
+
+  type _writeType = fileWriter(fmtType=JsonWriter);
+  type _readerT = fileReader(fmtType=JsonReader);
+  record JsonWriter {
+    var firstField = true;
+    var _inheritLevel = 0;
+
+    // TODO: rewrite in terms of writef, or something
+    proc _oldWrite(ch: _writeType, const val:?t) throws {
+      var _def = new DefaultWriter();
+      var dc = ch.withFormatter(_def);
+      var st = dc._styleInternal();
+      var orig = st; defer { dc._set_styleInternal(orig); }
+      st.realfmt = 2;
+      st.string_format = iostringformat.json:uint(8);
+      st.aggregate_style = QIO_AGGREGATE_FORMAT_JSON:uint(8);
+      st.array_style = QIO_ARRAY_FORMAT_JSON:uint(8);
+      st.tuple_style = QIO_TUPLE_FORMAT_JSON:uint(8);
+      dc._set_styleInternal(st);
+      dc._writeOne(dc.kind, val, here);
+    }
+
+    proc encode(writer: _writeType, const x:?t) throws {
+      if t == string  || isEnumType(t) || t == bytes {
+        writer.writeLiteral('"');
+        writer.withFormatter(DefaultWriter).write(x);
+        writer.writeLiteral('"');
+      } else if isNumericType(t) || isBoolType(t) {
+        _oldWrite(writer, x);
+      } else if t == ioLiteral {
+        writer._writeLiteral(x.val);
+      } else if t == ioNewline {
+        // Need to account for 'skipWhitespaceOnly' somehow...
```

Comment:
maybe put TODO at the start of that comment to make it clearer it is not done already